src/rust/engine/fs/src/store.rs

```diff
@@ -2001,23 +2025,28 @@ mod remote {
       request: bazel_protos::remote_execution::FindMissingBlobsRequest,
     ) -> impl Future<Item = HashSet<Digest>, Error = String> {
       let store = self.clone();
-      self.with_cas_client(move |client| {
-        client
-          .find_missing_blobs_opt(&request, store.call_option())
-          .map_err(|err| {
-            format!(
-              "Error from server in response to find_missing_blobs_request: {:?}",
-              err
-            )
-          })
-          .and_then(|response| {
-            response
-              .get_missing_blob_digests()
-              .iter()
-              .map(|digest| digest.into())
-              .collect()
-          })
-      })
+      let metadata =
+        ContentRequestMetadata::new(format!("finding missing blobs for {:?}", request));
```

Comment:
It may be worth giving `ContentRequestMetadata` an `Option<Box<Message>>` or `<Option<Box<Debug>>` which it can conditionally format if this string is actually needed, rather than needing to force a formatting here - proactively formatting here may actually be a little expensive and memory-intensive...